internal/features/features.go

```diff
@@ -49,6 +49,11 @@ const (
 	// https://github.com/crossplane/crossplane/blob/863ff6/design/design-doc-composition-functions.md
 	EnableBetaCompositionFunctions feature.Flag = "EnableBetaCompositionFunctions"
 
+	// EnableBetaCompositionFunctionsExtraResources enables extra resources support for
+	// composition functions. See the below design for more details.
+	// https://github.com/crossplane/crossplane/blob/863ff6/design/design-doc-composition-functions.md
+	EnableBetaCompositionFunctionsExtraResources feature.Flag = "EnableBetaCompositionFunctionsExtraResources"
```

Comment:
We were saying to keep a separate flag to allow disabling extra resource gathering for more security aware users, so I think it would make sense to keep it at least for now just to avoid having to add it back later as soon as someone asks for it.

Reply:
Oh right, I forgot about that.

I'm not sure a feature flag is the best fit for it though, in the sense that I don't think of it as being a lifecycle/maturity thing. i.e. I don't think this is being a flag due to its maturity, but rather because folks might want to turn it off even after functions (and extra resources) are GA.

That said, using a feature flag is effectively an implementation detail for now, so we can worry about this later. I imagine you'd toggle it with `--enable-composition-functions-extra-resources` whether it was a feature flag (in the alpha/beta/GA lifecycle sense) or not.

Reply:
yes, exactly, I ended up using the feature flag because it was already wired properly and it was near to the other flag it is linked to, it's indeed `--enable-composition-functions-extra-resources` and once we promote to beta we can just change the implementation.